File: src/main/java/fr/adrienbrault/idea/symfony2plugin/action/NewYamlServiceAction.java
package fr.adrienbrault.idea.symfony2plugin.action;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.PlatformDataKeys;
import com.intellij.openapi.project.Project;
import fr.adrienbrault.idea.symfony2plugin.Symfony2ProjectComponent;
import org.jetbrains.annotations.NotNull;

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
public class NewYamlServiceAction extends AbstractProjectDumbAwareAction {

    public NewYamlServiceAction() {
        super("Yaml Service", "Create new Yaml File", AllIcons.Nodes.DataTables);
    }

    @Override
    public void update(@NotNull AnActionEvent event) {
        this.setStatus(event, false);
        Project project = getEventProject(event);
        if (!Symfony2ProjectComponent.isEnabled(project)) {
            return;
        }

        if (NewFileActionUtil.getSelectedDirectoryFromAction(event) != null) {
            this.setStatus(event, true);
        }
    }

    @Override
    public void actionPerformed(AnActionEvent event) {
        final Project project = event.getData(PlatformDataKeys.PROJECT);
        ServiceActionUtil.buildFile(event, project, "/fileTemplates/container.yml");
    }
}

File: src/main/kotlin/fr/adrienbrault/idea/symfony2plugin/action/NewYamlServiceAction.kt
package fr.adrienbrault.idea.symfony2plugin.action

import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.AnActionEvent
import fr.adrienbrault.idea.symfony2plugin.Symfony2ProjectComponent

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
class NewYamlServiceAction : AbstractProjectDumbAwareAction(
    "Yaml Service",
    "Create new Yaml File",
    AllIcons.Nodes.DataTables
) {
    override fun update(event: AnActionEvent) {
        setStatus(
            event,
            Symfony2ProjectComponent.isEnabled(getEventProject(event)) &&
                NewFileActionUtil.getSelectedDirectoryFromAction(event) != null
        )
    }

    override fun actionPerformed(event: AnActionEvent) {
        val project = event.project ?: return
        ServiceActionUtil.buildFile(event, project, "/fileTemplates/container.yml")
    }
}
